Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsZipPreprocessExecuteAndRunUnitTestsWithDOMAndMochaExecutionStrategy.cs

```diff
@@ -0,0 +1,322 @@
+﻿namespace OJS.Workers.ExecutionStrategies
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using Checkers;
+    using Common;
+    using Executors;
+    using Ionic.Zip;
+    using OJS.Common;
+    using OJS.Common.Extensions;
+
+    public class NodeJsZipPreprocessExecuteAndRunUnitTestsWithDomAndMochaExecutionStrategy :
+        IoJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy
+    {
+        protected const string AppJsFileName = "app.js";
+
+        protected const string SubmissionFileName = "_$submission";
+
+        public NodeJsZipPreprocessExecuteAndRunUnitTestsWithDomAndMochaExecutionStrategy
+            (
+            string nodeJsExecutablePath,
+            string mochaModulePath,
+            string chaiModulePath,
+            string jsdomModulePath,
+            string jqueryModulePath,
+            string handlebarsModulePath,
+            string sinonModulePath,
+            string sinonChaiModulePath,
+            string underscoreModulePath,
+            string browserifyModulePath,
+            string babelifyModulePath,
+            string ecmaScriptImportPluginPath,
+            int baseTimeUsed,
+            int baseMemoryUsed)
+            : base(
+                nodeJsExecutablePath, mochaModulePath, chaiModulePath, jsdomModulePath, jqueryModulePath,
+                handlebarsModulePath, sinonModulePath, sinonChaiModulePath, underscoreModulePath, baseTimeUsed,
+                baseMemoryUsed)
+        {
+            if (!Directory.Exists(browserifyModulePath))
+            {
+                throw new ArgumentException($"Browsrify not found in: {browserifyModulePath}",
+                    nameof(browserifyModulePath));
+            }
+
+            if (!Directory.Exists(babelifyModulePath))
+            {
+                throw new ArgumentException($"Babel not found in: {babelifyModulePath}", nameof(babelifyModulePath));
+            }
+
+            if (!Directory.Exists(ecmaScriptImportPluginPath))
+            {
+                throw new ArgumentException($"ECMAScript2015ImportPluginPath not found in: {ecmaScriptImportPluginPath}",
```

Comment:
`$"ECMAScript2015ImportPluginPath not found in: {ecmaScriptImportPluginPath}",` -> new line